### DISCUSSION TITLE
Return True or False after sending email

### DISCUSSION THREAD

**Discussion by grim-reapper** [laravel/framework#34048] [General]
Hi, I'm requesting new features which is very important that is missing from laravel. The feature is related to laravel Mail::send() function. Laravel Mail::send() function does not return anything but null on sending email. So why i am requesting this features because in most we have to store email logs where we can set either email is sent or not base in Mail::send() function whereas Mail function doesent return anything so it's not possible to determine either email is sent or not. I have used PHPMailer library which is good mail library. PHPMailer library send() method return true or false on sending email which is very helpful. So could you please add this little feature? It would be very helpful for those who want to perform some action base on mail true false. 
I'm hoping my request will be considered.

**Comment by mfn** [comment]
For one, there's `\Illuminate\Contracts\Mail\Mailer::failures`
> Get the array of failed recipients

But to the best of my knowledge, it depends on the actual transport to properly support this.

AFAIK with SMTP it won't work as emails are usually enqueued and _bounce back_ so you don't get back a synchronous feedback.

But I know that other transport like postmarkapp using their API, they report many errors synchronously back before spooling them internally.

YMMV

**Reply by grim-reapper** [reply]
Yes, i know about this function. But i have gone through the core mailer file of framework where send function is written The code of that function is 
```
public function send($view, array $data = [], $callback = null)
    {
        if ($view instanceof MailableContract) {
            return $this->sendMailable($view);
        }
        [$view, $plain, $raw] = $this->parseView($view);

        $data['message'] = $message = $this->createMessage();

        $callback($message);

        $this->addContent($message, $view, $plain, $raw, $data);

        if (isset($this->to['address'])) {
            $this->setGlobalToAndRemoveCcAndBcc($message);
        }

        $swiftMessage = $message->getSwiftMessage();

        if ($this->shouldSendMessage($swiftMessage, $data)) {
            $this->sendSwiftMessage($swiftMessage);

            $this->dispatchSentEvent($message, $data);
        }
    }
```
if you look at the function it returns nothing, So, to send a message it calls following method inside `$this->sendSwiftMessage($swiftMessage);`
this method is actually used for sending email, if you explore this method it's return something, whereas in the above method where it's called nothing return back. So if the above function return the output of this method `$this->sendSwiftMessage($swiftMessage);` we can determine either email sent or not. so my request is just return the output of this `$this->sendSwiftMessage($swiftMessage);` function in above send method, it would be very helpful.

**Reply by mfn** [reply]
> , if you explore this method it's return something, whereas in the above method where it's called nothing return back

Right, but it doesn't need to necessary, if you look at https://github.com/laravel/framework/blob/1331ba81d71192172764e5786ce36f4967762c1a/src/Illuminate/Mail/Mailer.php#L516-L525

`$this->failedRecipients` is used _by reference_ from `\Swift_Mailer::send`.

What `send` returns is the number of recipient it sent it to, but `failedRecipients` contains the ones it couldn't (given my exception above too).

So, whilst I'm not arguing _against_ changing / improving the return value, I'm not sure if it changes anything for your goal, as you already have access to what you need. IHMO 😄
